### PR TITLE
:bug: Ignore ireturn, and varnamelen linters

### LINKED ISSUE #98: Check linter warnings/errors
There's a few new warnings that started to pop up with recent `golangci-lint`. Lets take a look if those are needed to be fixed or disabled.

https://github.com/knative-sandbox/kn-plugin-event/runs/4102850366?check_suite_focus=true

### PR DESCRIPTION
**Pull request by cardil**
# Changes

- :bug: Ignore _ireturn_, and _varnamelen_ linters

/kind bug

Fixes #98 

## Justification

I did a closer look at those two new rules that came with the latest release of golangci-lint: [ireturn](https://github.com/butuzov/ireturn) and [varnamelen](https://github.com/blizzy78/varnamelen), and I believe it's not worth pursuing compliance with them. Detailed evaluation below.

### _ireturn_ evaluation

The _ireturn_ linter ensures you don't return interfaces, but rather a concrete type. In general, that's good practice when the struct is public - the caller might be interested in struct fields, not only the behavior (interface). But, _ireturn_ fires also in case when the returned type is private - linter even advise such change in its README doc. Doing that violates another linter - revive, and it's cumbersome to caller, not giving him any benefits from it.

There are other cases that seem reasonable to skip, like: 

 * [factory methods #21](https://github.com/butuzov/ireturn/issues/21),
 * [chain of responsibility pattern #22](https://github.com/butuzov/ireturn/issues/22)
 * [private structs with single interface #31](https://github.com/butuzov/ireturn/issues/31)

Also, worth citing Rob Pike on this https://github.com/go-proverbs/go-proverbs.github.io/issues/37#issuecomment-571345739

### _varnamelen_ evaluation

The _varnamelen_ forces use of names size 3+ for most of the variables. It applies to all places where the function is longer than 5 lines, so almost all places. That makes it really hard to use, as Go code tends to have short packages, like `git`, `file`, `metadata`, `plugin`, and so 3+ length variable names tend to interfere a lot with those packages.

I've experimented a bit with different settings:

```yaml
# The longest distance, in source lines, that is being considered a "small scope." (defaults to 5)
# Variables used in at most this many lines will be ignored.
max-distance: 12
# The minimum length of a variable's name that is considered "long." (defaults to 3)
# Variable names that are at least this long will be ignored.
min-name-length: 2
```

And still, I got 10 places that are not compliant, in such small project. Manually reviewing them, I don't see a huge benefit of changing those names. It doesn't improve readability IMHO.

### CODE REVIEW

File: .github/workflows/knative-style.yaml [hunk 1]
```diff
@@ -106,7 +106,7 @@ jobs:
         if: steps.golangci_configuration.outputs.files_exists == 'true'
         uses: golangci/golangci-lint-action@v2
         with:
-          version: v1.42
+          version: v1.43
```
Comment (cardil) [inline .github/workflows/knative-style.yaml:109]: This change is taken from #99

Comment (dsimansk) [reply .github/workflows/knative-style.yaml:109]: I hope that won't break update actions :see_no_evil:

**Review by dsimansk — APPROVED** [review]
/approve
/lgtm

File: .github/workflows/knative-style.yaml (hunk 1 above)
Comment (cardil) [reply .github/workflows/knative-style.yaml:109]: I think it won’t.

**Pull request merged**